### DISCUSSION TITLE
Question about continue running

### DISCUSSION THREAD

**Discussion by naruto1980** [pybricks/support#1943] [Q&A] [answered]
First of all, I would like to special shout-out for Pybricks. Our team passed the regional round as first place. We use beta python code and it is very stable. My daughter also love Pybricks very much compare with block coding. She feels easier to code.

My question is regarding the continue running of robot without stop after every session (straight, turn), I tried to use then=Stop.NONE but look like it does not work. Or should I slow down the speed to make it run smoother?

Below is the code I use
```
drive_base.settings(straight_speed=FAST_SPEED,turn_rate=100)
drive_base.straight(400, then=Stop.NONE)
drive_base.turn(45, then=Stop.NONE)
drive_base.straight(150, then=Stop.NONE)
drive_base.turn(-45, then=Stop.NONE)
drive_base.straight(330)
```

https://github.com/user-attachments/assets/f46d7f3a-5fc0-4368-8f68-c73869d42b5d

**Comment by laurensvalk** [comment, marked as answer]
> First of all, I would like to special shout-out for Pybricks. Our team passed the regional round as first place. We use beta python code and it is very stable. My daughter also love Pybricks very much compare with block coding. She feels easier to code.

Thank you!

>  I tried to use then=Stop.NONE (...)

We try to be on the edge of what's possible, but we're still bound by the laws of physics :smile: 

`then=Stop.NONE` works best when you combine `straight` with a `curve`. Then it can keep its forward speed as it makes the curve.

With `turn`, it will try to turn in place, and thus abruptly slow down to achieve that.

**Reply by naruto1980** [reply]
I tried it and it works. It also depend on the wheel which make it turn smoother. 
Thank you very much.
